### ISSUE TITLE
HmIP-DRSI4

### ISSUE THREAD

**Issue by slowflyer** [coreGreenberet/homematicip-rest-api#387]
Hi,
I want to turn light's on and off by the API using the HomematicIP "LINKED_SWITCHING" group "Panic"
Therefore I modified api.py from the samples.
Right now I get the devices, but if the device is a DinRailSwitch4, I do not get the channel of the light used in the Panic group.
Is it currently not possible to get the channel or am I overseeing something?

```
import homematicip
from homematicip.home import Home
from homematicip.device import BrandSwitchMeasuring, DinRailSwitch4, DinRailSwitch, Switch

config = homematicip.find_and_load_config_file()

home = Home()
home.set_auth_token(config.auth_token)
home.init(config.access_point)

def write_shutter(label,room,device):
        print(label,") ", room, " ", device.label, " ", device.lastStatusUpdate, " ")
        print(device.on)

def write_plugableswitchmeasuring(room,device):
        print(room, " ", device.label, " ", device.lastStatusUpdate, " ", device.currentPowerConsumption, " ", device.energyCounter)

def write_heatingthermostat(room,device):
        print(room, " ", device.label, " ", device.lastStatusUpdate)

def write_wallmountedthermostatpro(room,device):
    print(room, " ", device.label, " ", device.lastStatusUpdate, " ", device.actualTemperature, " ", device.setPointTemperature, " ", device.humidity)

def main():
    global home
    home.get_current_state()
    for g in home.groups:
        if g.groupType=="LINKED_SWITCHING" and g.label == 'PANIC':
            for d in g.devices:
                if isinstance(d,Switch):
                    write_shutter('a',g.label,d)
                    print(d)
                    d.turn_on()
                elif isinstance(d,DinRailSwitch4):
                    write_shutter('b',g.label,d)
                    print(d)
                    d.turn_off()
                elif isinstance(d,DinRailSwitch):
                    write_shutter('c',g.label,d)
                elif isinstance(d,BrandSwitchMeasuring):
                    write_shutter('d',g.label,d)

if __name__ == "__main__":
    main()
```

Thanks,
slowflyer

**Comment by coreGreenberet** [comment]
DinRailSwitch is a subclass of Switch, so the first "if isinstance(d, Switch):" will be true for a DinRailSwitch and because of your elif it will never his the other blocks. 
Just reorder your blocks and put isinstance(d, Switch) als the last block

**Comment by slowflyer** [comment]
Thanks for your feedback. I originally had DinRailSwitch4 first. But that did'nt make a difference. When print out my DinRailSwitch4 device object it looks like:

HmIP-DRSI4 Schaltaktor 4K1 4K2 4K3 4Kx lowBat(None) unreach(False) rssiDeviceValue(128) rssiPeerValue(-61) configPending(False) dutyCycle(False) deviceOverheated(False) devicePowerFailureDetected(False) on(None) profileMode(None) userDesiredProfileMode(None)

It contains no information about the channel. And without the information about the channel, I can't turn that device off.

**Comment by coreGreenberet** [comment]
the str method will only print the first channel of the device. The DRSI4 has 4 channels, to look it up you would have to access the correct functionalChannel

**Comment by slowflyer** [comment]
Ok with:
`        for c in device.functionalChannels:
            if isinstance(c,MultiModeInputSwitchChannel):
                print(c.label)
                print(c.on)
                print(c.index)`

I get the channels of my DinRailSwitch4, but at this point I do not get any information if this channel belongs to the group I'm looking for.

**Comment by coreGreenberet** [comment]
that got fixed #388 and should be in the next release
